app/models/service_template.rb

```diff
@@ -55,6 +55,8 @@ class ServiceTemplate < ApplicationRecord
 
   has_many   :dialogs, -> { distinct }, :through => :resource_actions
 
+  has_many   :miq_requests, ->(template) { where(["source_id = ? AND source_type = 'ServiceTemplate'", template.id]).order(:created_on) }, :foreign_key => "source_id"
```

Comment:
Is `.order(:created_on)` really required here?  I feel like this should be left to the caller to append if needed.

Also, does this need to be accessible from the API?  cc @abellotti

Reply:
I can remove the sorting. Was following examples from other model.
Checked with @abellotti. No changed is needed from API side.

Reply:
Isn't this as simple as `has_many :miq_requests, :as => :source`?
http://guides.rubyonrails.org/association_basics.html#polymorphic-associations

Reply:
yeah, this feels like revinventing Rails.